--- frama-c-Sulfur-20171101/src/plugins/e-acsl/mmodel_analysis.ml.orig  0000-00-00 00:00:00.000000000 +0000
+++ frama-c-Sulfur-20171101/src/plugins/e-acsl/mmodel_analysis.ml       2017-12-27 13:35:27.035904837 +0000
@@ -710,7 +710,7 @@
     Options.fatal "[pre_analysis] unexpected expression %a" Exp.pretty e
 
 (* ************************************************************************** *)
-(** {Public API} {{{ *)
+(* Public API {{{ *)
 (* ************************************************************************** *)
 
 let must_model_vi ?bhv ?kf ?stmt vi =
@@ -730,6 +730,8 @@
 let use_model () = not (Env.is_empty ()) || Options.Full_mmodel.get () ||
   Env.has_heap_allocations ()
 
+(* }}} *)
+
 (*
 Local Variables:
 compile-command: "make"
